lib/database/repositories/EnvironmentHistoryItemRepository.js

```diff
@@ -11,8 +11,35 @@
  * or submit itself to any jurisdiction.
  */
 
-const { models: { EnvironmentHistoryItem } } = require('../');
+const { models: { EnvironmentHistoryItem }, sequelize } = require('../');
 const Repository = require('./Repository');
+const { timestampToMysql } = require('../../server/utilities/timestampToMysql.js');
+
+/**
+ * Returns the SQL query for fetching the environment history combinations and
+ * their occurrences within a specified time period.
+ *
+ * @param {Period} period the period on which the occurrences must be counted
+ * @return {string} the raw SQL query
+ */
+const getHistoryDistributionQuery = ({ from, to }) => `
+    SELECT environments, COUNT(*) AS count
+    FROM (
+        SELECT e.id, GROUP_CONCAT(ehi.status) AS environments
```

Comment:
I would rather name it `statusHistory` than `environment`

Reply:
Again you are entirely right, the only reason why I put it like this, is because of the slides that refer to statusHistory as environments.